### PR TITLE
Upgrade to Kotlin 1.6.10

### PR DESCRIPTION
**Pull request by reconman**

### CODE REVIEW

**Review by MathewSachin — COMMENTED** [review]
I think this can be removed

https://github.com/Fate-Grand-Automata/FGA/blob/94fcde26add75f6ec594f526f658c4ec63621640/build.gradle#L43

**Comment by reconman** [comment]
> I think this can be removed
> 
> https://github.com/Fate-Grand-Automata/FGA/blob/94fcde26add75f6ec594f526f658c4ec63621640/build.gradle#L43

Nope, still needed for other stuff like `delay()` and some time measuring.

I'm currently checking if any libraries can be updated other than kotlinx-coroutines.

File: prefs/build.gradle [hunk 1]
```diff
@@ -34,5 +34,5 @@ dependencies {
     implementation "com.google.dagger:hilt-android:$hilt_version"
     kapt "com.google.dagger:hilt-compiler:$hilt_version"
 
-    api 'com.github.tfcporciuncula:flow-preferences:1.3.1'
+    api 'com.github.tfcporciuncula:flow-preferences:1.4.0'
```
Comment (MathewSachin) [inline prefs/build.gradle:37]: I haven't checked the differences but there's already a v1.6.0

File: app/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -1,11 +1,14 @@
 <?xml version="1.0" encoding="utf-8"?>
 <manifest xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:tools="http://schemas.android.com/tools"
     package="com.mathewsachin.fategrandautomata">
 
     <uses-permission android:name="android.permission.FOREGROUND_SERVICE" />
     <uses-permission android:name="android.permission.VIBRATE" />
     <uses-permission android:name="android.permission.SYSTEM_ALERT_WINDOW" />
 
+    <uses-sdk tools:overrideLibrary="com.tfcporciuncula.flow.tests" />
```
Comment (MathewSachin) [inline app/src/main/AndroidManifest.xml:10]: I'm a bit confused on why we're having a dependency on their test library 🤔

File: prefs/build.gradle (hunk 1 above)
Comment (MathewSachin) [reply prefs/build.gradle:37]: Looks like they've changed the package ID

https://github.com/tfcporciuncula/flow-preferences

**Pull request merged**